store/sqlstore/plugin_store.go

```diff
@@ -111,6 +112,29 @@ func (ps SqlPluginStore) CompareAndSet(kv *model.PluginKeyValue, oldValue []byte
 	return true, nil
 }
 
+func (ps SqlPluginStore) SetWithOptions(pluginId string, key string, value interface{}, options *model.PluginKVSetOptions) (bool, *model.AppError) {
+	if err := options.IsValid(); err != nil {
+		return false, err
+	}
+
+	kv, err := options.GetPluginKeyValue(pluginId, key, value)
+	if err != nil {
+		return false, err
+	}
+
+	if options.Atomic {
+		serializedOldValue, err := options.GetOldValueSerialized()
+		if err != nil {
+			return false, err
+		}
+
+		return ps.CompareAndSet(kv, serializedOldValue)
+	} else {
```

Comment:
this else can be dropped

https://yourbasic.org/golang/if-else-statement/
https://groups.google.com/forum/#!topic/golang-nuts/G92rQDoCG5g